### PR TITLE
feat(client): show the default 404 page only when profile fails

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Because of the removal of the timer, the page render before the `maybeUser` data is fetched through the useEffect, this aims to delay it, but Idk if it actually will work because it isn't possible to test :(

You can find the issue if you visit https://www.freecodecamp.org/Sboonny in private mode

### CODE REVIEW

File: client/src/client-only-routes/show-profile-or-four-oh-four.tsx [hunk 1]
```diff
@@ -75,8 +75,9 @@ function ShowProfileOrFourOhFour({
   if (!isBrowser()) {
     return null;
   }
+  const hasPorfile = isEmpty(requestedUser) || isEmpty(maybeUser);
```
Comment (naomi-lgbt) [inline client/src/client-only-routes/show-profile-or-four-oh-four.tsx:78]:
```suggestion
  const hasProfile = isEmpty(requestedUser) || isEmpty(maybeUser);
```

File: client/src/client-only-routes/show-profile-or-four-oh-four.tsx [hunk 2]
```diff
@@ -75,8 +75,9 @@ function ShowProfileOrFourOhFour({
...
 
-  return isEmpty(requestedUser) ? (
+  return hasPorfile ? (
```
Comment (naomi-lgbt) [inline client/src/client-only-routes/show-profile-or-four-oh-four.tsx:80]:
```suggestion
  return hasProfile ? (
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

**Comment by naomi-lgbt** [comment]
Is the aim to fix the issue where the default 404 page shows until the profile is loaded?

If so, testing this locally doesn't appear to resolve it.

**Comment by Sboonny** [comment]
how did you test it? I couldn't mimic the way to produce the bug with the way that 404 is structured

**Comment by naomi-lgbt** [comment]
- Open the network tab, throttle down to 3G speed to get the slow loading times.
- Visit `http://localhost:8000/twaha` - wait for it to load (takes forever)
- Click `Preview custom 404 page`

You should see the 404 pop up for a bit before switching to twaha's private profile.

**Comment by Sboonny** [comment]
Nice, I can finally test, I should be able to sort it tomorrow morning

**Comment by Sboonny** [comment]
I have wrapped error boundary around profile, so when it fails and we get runtime error, we display 404 page.

**Comment by naomi-lgbt** [comment]
This is looking good. Just to confirm, are we okay with this error message displaying?

image

**Comment by Sboonny** [comment]
I can sort it in another PR, because I want to refactor the loading component as well

**Review by naomi-lgbt — APPROVED** [review]
Okie dokie

**Review by Nirajn2311 — APPROVED** [review]

**Comment by raisedadead** [comment]
> This is looking good. Just to confirm, are we okay with this error message displaying?

Hold up! 

I need to get up to speed on these changes, but that error message and screen were there specifically when the client got stuck loading. This can happen when we release a new bundle, and the client has a stale cache. 

Ex: You are working on your curriculum, we make a release, and then you submit challenges. Ideally, you do not see the difference, but depending on your ISP, DNS, etc., the new client bundle could take a bit to stream to your browser. 

In those cases, the client can't reach the API to update states and would need a quick reload.

The client (or rather the user) will not know about this unless they see a message that shows up after a pre-determined period.

**Comment by Sboonny** [comment]
> that error message and screen were there specifically when the client got stuck loading. This can happen when we release a new bundle, and the client has a stale cache.

this behavior hasn't changed, what changed is the way we display this loading screen for the profile page.

**Comment by Sboonny** [comment]
The reason I want to refactor this message is that Naomi shouldn't have never seen this message, because there isn't a message delay as far as I know.

There is a misunderstanding from my part on that component and I want to make it simpler and more easier to understand.

**Pull request merged**